src/lib.rs

```diff
@@ -173,7 +175,10 @@ pub fn build(
 /// Executes the build artifacts cleaning.
 ///
 pub fn clean() -> anyhow::Result<()> {
-    Ok(std::fs::remove_dir_all(PathBuf::from(
-        LLVMPath::DIRECTORY_LLVM_TARGET,
-    ))?)
+    match std::fs::remove_dir_all(PathBuf::from(LLVMPath::DIRECTORY_LLVM_TARGET)) {
+        Ok(_) => Ok(()),
+        Err(err) => {
+            anyhow::bail!("unable to remove LLVM directory: {}", err);
```

Comment:
Please start error messages capitalized.

Reply:
You can use the `?` operator instead of `match` here.

Reply:
Good point, I've updated the code to use `?` operator and updated the error handling using `with_context` function as [anyhow](https://docs.rs/anyhow/latest/anyhow/) documentation suggests as a best practice.

Reply:
`with_context` had been broken for a long time. Last time I checked, it prevented the error message from being printed correctly.
Please check, and revert to `.map_err` if the issue is still present.

Reply:
https://github.com/matter-labs/era-compiler-llvm-builder/pull/13#discussion_r1488399078